tree/treeplayer/src/TTreePlayer.cxx

```diff
@@ -2356,7 +2356,8 @@ void TTreePlayer::RecursiveRemove(TObject *obj)
 ///
 /// You can change the default number of rows to be shown before `<CR>`
 /// via  mytree->SetScanField(maxrows) where maxrows is 50 by default.
-/// if maxrows is set to 0 all rows of the Tree are shown.
+/// if maxrows is set to 0 all rows of the Tree are shown, and you are
+/// not prompted to press `<CR>` or `q` to end the loop.
```

Comment:
I see also here the addition of this extra detail, probably I am not seeing the same issue as you are. Could you elaborate on why you believe it is important to specify that the user does not need to press an extra key when all the rows are printed?

Reply:
My use case was: I want to use Scan, but I do not want to press a key later on interactively at the prompt.
So first thing I tried was to specify lastEntry = totalEntries as Scan parameter, but still it wouldn't exit, I had still to press CR or q after that.
It took me a while to figure out that SetScanField(0) removes the interactivity, even if the print output is the same as Scan("", totalEntries);

Reply:
Thanks! Now I better understand the reasoning behind the changes. Just for completeness, I have tried calling with the syntax
```
tree->Scan("myColumn", /*selection*/ "", /*option*/ "", /*nentries*/ 5)
```
And it printed 5 entries to the prompt and then returned back without having to press the extra key stroke. Would this also represent another way to achieve what you wanted?

Reply:
weird... I get different results.
2 entries: as you said
3 entries: as you said
4 entries: I get almost 50 printed and then <CR> / q prompt
So seems buggy... or depends on how many lines your terminal has maybe?

Reply:
Ok, understood, there were several instances of the same row.

Reply:
In other words, there is no way to ensure that you are not prompted just by restricting with nentries, as there might be 100 rows for one entry.

Reply:
To illustrate from ROOT-8256 JIRA root file:

root -l -b -q -e 'TChain c("Nominal/llllTree"); c.Add("my.output.root"); c.SetScanField(4); c.Scan("llll_type","","",4);'

This works well and shows 4 lines and CR then.

root -l -b -q -e 'TChain c("Nominal/llllTree"); c.Add("my.output.root"); c.Scan("llll_type","","",4);'

So only way to be sure is SetScanField(0);

Reply:
Ah yes because `TTree::Scan` unrolls the collections to multiple rows! Understood now, thanks!